test/azure_test.js

```diff
@@ -181,4 +180,133 @@ suite('azure table (sas)', function() {
+    assert(new Date(result.expiry).getTime() > new Date().getTime(),
+      "Expected expiry to be in the future");
+
+    let blob = new azure.Blob({
+      accountId: helper.testaccount,
+      sas: result.sas
+    });
+
+    result = await blob.putBlob('container-test', 'blobTest', {type: 'BlockBlob'});
+    assert(result);
+  });
+
+  test('azureBlobSAS (read-only)', async () => {
+    let result = await helper.auth.azureBlobSAS(
+      helper.testaccount,
+      'container-test',
+      'read-only',
+    );
+    assert(typeof(result.sas) === 'string', "Expected some form of string");
+    assert(new Date(result.expiry).getTime() > new Date().getTime(),
+      "Expected expiry to be in the future");
+
+    let blob = new azure.Blob({
+      accountId: helper.testaccount,
+      sas: result.sas
+    });
+
+    try {
+      await blob.putBlob('container-test', 'blobTest', {type: 'BlockBlob'});
+    } catch (error) {
+      assert.equal(error.code, 'AuthorizationPermissionMismatch');
+      return;
+    }
+    assert(false, 'This should have thrown an error because the write is not allowed.');
+  });
+
+  test('azureBlobSAS (invalid level)', async () => {
+    try {
+      await helper.auth.azureBlobSAS(
+        helper.testaccount,
+        'container-test',
+        'foo-bar-baz',
+      );
+    } catch (error) {
+      assert.equal(error.code, 'InvalidRequestArguments');
+      return;
+    }
+    assert(false, "This should have thrown an error");
+  });
+
+  test('azureBlobSAS (allowed container)', async () => {
+    let auth = new helper.Auth({
+      baseUrl:          helper.baseUrl,
+      credentials:      rootCredentials,
+      authorizedScopes: [
+        'auth:azure-blob:read-write:' + helper.testaccount + '/allowed-container'
+      ]
+    });
+
+    let result = await auth.azureBlobSAS(
+      helper.testaccount,
+      'allowed-container',
+      'read-write'
+    );
+
+    assert(typeof(result.sas) === 'string', "Expected some form of string");
+    assert(new Date(result.expiry).getTime() > new Date().getTime(),
+      "Expected expiry to be in the future");
+  });
+
+  test('azureBlobSAS (allowed read-write -> read-only)', async () => {
+    var auth = new helper.Auth({
+      baseUrl:          helper.baseUrl,
+      credentials:      rootCredentials,
+      authorizedScopes: [
+        'auth:azure-blob:read-write:' + helper.testaccount + '/allowed-container'
+      ]
+    });
+
+    let result = await auth.azureBlobSAS(
+      helper.testaccount,
+      'allowed-container',
+      'read-only',
+    );
+    assert(typeof(result.sas) === 'string', "Expected some form of string");
+    assert(new Date(result.expiry).getTime() > new Date().getTime(),
+      "Expected expiry to be in the future");
+  });
+
+  test('azureBlobSAS (unauthorized container)', async () => {
+    let auth = new helper.Auth({
+      baseUrl:          helper.baseUrl,
+      credentials:      rootCredentials,
+      authorizedScopes: [
```

Comment:
oh, we don't have it here...
In queue we do: https://github.com/taskcluster/taskcluster-auth/blob/ce1f509c11179875674a7c548ca07fd0e8db20ac/test/helper.js#L76-L82

We could do the same here: 
https://github.com/taskcluster/taskcluster-auth/blob/ce1f509c11179875674a7c548ca07fd0e8db20ac/test/helper.js#L76-L82

Ie. basically wrap with  a `helper.scopes` method...